latex_access/tests/test_preprocessor.py

```diff
@@ -0,0 +1,70 @@
+import unittest
+import tempfile
+import pickle
+try:
+    from latex_access import preprocessor
+except ModuleNotFoundError:
+    incompatible = True
+else:
+    incompatible = False
+
+
+@unittest.skipIf(incompatible, "This module is not compatible with version 3 of Python.")
+class TestPreprocessor(unittest.TestCase):
+    def setUp(self):
+        """Creates a preprocessor instance."""
+        self.preprocessor = preprocessor.preprocessor()
+
+    def test_table(self):
+        """Tests creation of table attribute."""
+        self.assertEqual(self.preprocessor.table, {})
+
+    def test_add(self):
+        """Test adding command to table."""
+        self.preprocessor.add('foo', 'bar')
+        self.assertEqual(self.preprocessor.table['foo'], ('bar'))
+
+    def test_add_from_string(self):
+        """Tests adding command with args."""
+        self.preprocessor.add_from_string('foo', 'args', '1#2#3')
+        self.assertEqual(self.preprocessor.table['foo'], ['args', '1', 2, '', 3, ''])
+
+    def test_write(self):
+        """Tests that preprocessor entries can be saved to a file."""
+        file = tempfile.NamedTemporaryFile(delete=False)
+        self.preprocessor.table['foo'] = 'bar'
+        self.preprocessor.write(file.name)
+        f = open(file.name)
+        temp_table = pickle.load(f)
+        f.close()
+        self.assertEqual(temp_table['foo'], 'bar')
+
+    def test_read(self):
+        """Tests that preprocessor entries can be read from a file."""
+        file = tempfile.NamedTemporaryFile(delete=False)
+        temp_table = {'foo': 'bar'}
```

Comment:
I'm afraid this test should be rewritten. We want to check if the pickle file created by the preprocessor can be read back, so rather than pickling a normal dictionary just create a file using preprocessor in advance, add it to the version control and check if entries from it can be loaded. This would become pretty important after migrating to Python 3 to make sure that if someone saved their preprocessor entries they would still work after migration.